ignite/metrics/gan/fid.py

```diff
@@ -136,22 +119,26 @@ def __init__(
         except ImportError:
             raise RuntimeError("This module requires scipy to be installed.")
 
+        if num_features is not None and num_features <= 0:
+            raise ValueError(f"Argument num_features must be greater to zero, got: {num_features}")
+
         # default is inception
+        self._num_features, self._feature_extractor = self._check_input(num_features, feature_extractor, device)
+        self._eps = 1e-6
+        super(FID, self).__init__(output_transform=output_transform, device=device)
+
+    @staticmethod
+    def _check_input(
+        num_features: Optional[int], feature_extractor: Optional[torch.nn.Module], device: Union[str, torch.device]
+    ) -> Tuple[int, torch.nn.Module]:
         if num_features is None and feature_extractor is None:
-            num_features = 2048
-            feature_extractor = InceptionExtractor()
+            return 2048, InceptionModel(return_features=True, device=device)
         elif num_features is None:
-            raise ValueError("Argument num_features should be defined")
+            raise ValueError("Argument num_features should be defined, if feature_extractor is provided")
         elif feature_extractor is None:
-            self._feature_extractor = lambda x: x
-            feature_extractor = self._feature_extractor
-
-        if num_features <= 0:
-            raise ValueError(f"Argument num_features must be greater to zero, got: {num_features}")
-        self._num_features = num_features
-        self._feature_extractor = feature_extractor
-        self._eps = 1e-6
-        super(FID, self).__init__(output_transform=output_transform, device=device)
+            return num_features, torch.nn.Identity().to(device)
+        else:
+            return num_features, feature_extractor.to(device)
```

Comment:
We need to ensure that `feature_extractor` is torch.nn.Module and raise TypeError otherwise.